lib/grape/dsl/inside_route.rb

```diff
@@ -29,7 +29,10 @@ module PostBeforeFilter
         def declared(params, options = {}, declared_params = nil)
           options = options.reverse_merge(include_missing: true, include_parent_namespaces: true)
 
-          declared_params ||= (!options[:include_parent_namespaces] ? route_setting(:declared_params) : (route_setting(:saved_declared_params) || [])).flatten(1) || []
+          all_declared_params = route_setting(:declared_params)
+          current_namespace_declared_params = route_setting(:saved_declared_params).last
```

Comment:
I tried to investigate the reason why these additional checks were introduced, but it's still unclear for me (PR #739 and specifically [here](https://github.com/ruby-grape/grape/pull/739/files#diff-e9e8424a5238d48301e313d8fe285697R22)).

I also played around endpoint structure to detect the case when `flatten` was really needed. But specs were green no matter what I did. I would personally prefer to clean up the code that does not break tests. What do you think?